app/services/analytics_service.rb

```diff
@@ -0,0 +1,102 @@
+class AnalyticsService
```

Comment:
I suggest namespacing `Analytics` and change the service name to be more specific. That'll make the code better structured for the future.

Reply:
I agree with that, but not sure how to name it properly. Were you thinking something like `AnalyticsService::Pro`?